docs: reorder kubernetes provider section

dosmanak opened the pull request:
/label Enhancement
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug
- Enhancement
- Breaking change
- Deprecation
- Cleanup
- Docs
-->

## What does this PR do?

Improve readability of kubernetes provider section of autodiscover functionality.


<!-- Mandatory
Explain here the changes you made on the PR. Please explain the WHAT: patterns used, algorithms implemented, design architecture, message processing, etc.
-->

## Why is it important?

I could not comprehend what are the configuration settings for kubernetes type of autodiscovery provider.
It seemd to me the only option to limit filebeat on namespace is using template.
<!-- Mandatory
Explain here the WHY, or the rationale/motivation for the changes.
-->

## Checklist

<!-- Mandatory
Add a checklist of things that are required to be reviewed in order to have the PR approved

List here all the items you have verified BEFORE sending this PR. Please DO NOT remove any item, striking through those that do not apply. (Just in case, strikethrough uses two tildes. ~~Scratch this.~~)
-->

- [ ] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
- [ ] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

elasticmachine commented:
## :grey_exclamation: Build Aborted
  
  > The PR is not allowed to run in the CI yet
  


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-29443/detail/PR-29443/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-29443/detail/PR-29443/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-29443/detail/PR-29443/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-29443/detail/PR-29443/1//artifacts)   [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/beats-ci/transactions/view?rangeFrom=2021-12-15T08:33:31.450Z&rangeTo=2021-12-15T08:53:31.450Z&transactionName=Beats%2Fbeats%2FPR-%7Bnumber%7D&transactionType=unknown&latencyAggregationType=avg&traceId=8acd136cf9bd2457b2e809bbb5b9e273&transactionId=89c5ac5f93404d02)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats



* Reason: `The PR is not allowed to run in the CI yet`


* Start Time: 2021-12-15T08:43:31.450+0000


* Duration: 5 min 46 sec


* Commit: 775eb1fc7d037d4671cc745ba5d345c138fc5d1c




</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY -->












## :robot: GitHub comments

To re-run your PR in the CI, just comment with:



- `/test` : Re-trigger the build.


- `/package` : Generate the packages and run the E2E tests.


- `/beats-tester` : Run the installation tests with beats-tester.


- `run` `elasticsearch-ci/docs` : Re-trigger the docs validation. (use unformatted text in the comment!)




<!--COMMENT_GENERATED_WITH_ID_comment.id-->

dosmanak commented:
/test

ruflin commented:
I retargeted this branch to 7.17 as I just created the 7.17 branch. It is always possible to backport this to 7.16 if needed for the docs. I'm wondering if this should even go into master first and backported from there. @dedemorton you and team likely know best.

dedemorton commented:
Yes, this should go into master then be backported to relevant branches. I'll put it on my list of things to review next week. Thanks!

elasticmachine commented:
Pinging @elastic/obs-docs (Team:Docs)

dosmanak commented:
@dedemorton Hi, I switched to master and cherry-picked the commit, but I noticed there are some changes, so I made the reorder manualy again and split it into 2 commits to distinguish my mind.
Thank you for looking into this!

dedemorton commented:
/test

dedemorton commented:
run elasticsearch-ci/docs

dosmanak commented:
I don't get why the build fails.
/test

dedemorton commented on libbeat/docs/shared-autodiscover.asciidoc:
```diff
@@ -136,6 +136,75 @@ endif::[]
+  making the provider to enable the provided templates only when it will gain the leader lease.
+  This setting can only be combined with `cluster` scope. When `unique` is enabled enabled, `resource`
+  and `add_resource_metadata` settings are not taken into account.
+`leader_lease`:: (Optional) Defaults to `{beatname_lc}-cluster-leader`. This will be name of the lock lease.
```
```suggestion
`leader_lease`:: (Optional) Defaults to +{beatname_lc}-cluster-leader+. This will be name of the lock lease.
```

dedemorton replied:
Found this existing nit in the content you moved. This change is needed so that {beatname_lc} resolves correctly.

dosmanak replied:
I do not undestand this templating system, so I accept your suggestion.

dosmanak replied:
Well, if I read the asciidoc reference well, the plus sign disables formating inside + symbols.
Here you replace emphasize with plus symbols which has totaly different function. I see you want to keep  {beatname_lc} unresolved at the moment.

dosmanak replied:
I see similar pattern used on line 8 of the file altought I do not understand how it complies with asciidocs standard.
Either way this bug was there even before.

dosmanak replied:
I am sorry, I read wrong documentation on asciidoctor, now I have fount the correct one.
https://asciidoc-py.github.io/userguide.html#X81

dedemorton reviewed (approved):
Thanks for your perseverance here. The errors are not related to anything you've done; they're likely caused by a flaky build. I like how you've reordered the content, but I think the same change should be made to the other sections on this page to make them parallel. I'm not going to ask you to do that, though, unless you feel inclined. We can deal with it later if you don't have time (this topic needs other work to make it easier to use, so no worries).

Since `kube_client_options` only appears in master (and not 7.17 or 7.16), we should backport this to 8.0 only. I'll make sure the right backport tags are applied.

Let me know if you want to reorder the other sections. If not, I can merge this.

dosmanak commented:
> Let me know if you want to reorder the other sections. If not, I can merge this.

I will try reorder also for other provider types and I will add it to this PR.

dosmanak commented:
> > Let me know if you want to reorder the other sections. If not, I can merge this.
> 
> I will try reorder also for other provider types and I will add it to this PR.

@dedemorton I appreciate CR, I did not change AWS providers hence it is too strange for me.
It seems the docker setting fell into warning section before. I am building the docs locally atm, so I will fix if I found anything.

dosmanak commented:
run elasticsearch-ci/docs

dedemorton commented:
Thanks for following through with additional changes! I've been on vacation, but will do a final review of this today.

dedemorton reviewed (approved):
LGTM. Thanks so much for your contribution!
